components/dropdown/user/src/index.js

```diff
@@ -55,55 +51,40 @@ class DropdownUser extends Component {
     )
   }
 
-  render() {
-    const {expanded, collapseByTouch} = this.state
-    const {
-      user,
-      menu,
-      expandOnMouseOver,
-      hasNotifications = this.props.menu.some(({notifications}) =>
-        Boolean(notifications)
-      )
-    } = this.props
-    const {name, avatar} = user
-    const wrapperClassName = cx('sui-DropdownUser', {
-      'is-expanded': expanded,
-      'has-notifications': hasNotifications
-    })
-    return (
+  const {expanded, collapseByTouch} = state
+  const {name, avatar} = user
+  const wrapperClassName = cx('sui-DropdownUser', {
+    'is-expanded': expanded,
+    'has-notifications': hasNotifications
+  })
+
+  return (
+    <div
+      className={wrapperClassName}
+      onMouseOver={expandOnMouseOver ? onMouseOver : NO_OP}
+      onMouseOut={expandOnMouseOver ? onMouseOut : NO_OP}
```

Comment:
Those evaluations can be done inside `onMouseOver` and `onMouseOut` in order to simplify render.